### PR TITLE
Send keepalive packet in utp multiplexer

### PR DESCRIPTION
**Pull request by canndrew**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/crust/543)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @vinipsmaker

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/lib.rs [hunk 1]
```diff
@@ -61,10 +61,11 @@ extern crate get_if_addrs;
 extern crate sodiumoxide;
 extern crate config_file_handler;
 extern crate service_discovery;
-
 #[cfg(target_family = "unix")]
 extern crate libc;
 extern crate nat_traversal;
+#[cfg(test)]
+extern crate void;
```
Comment (Fraser999) [inline src/lib.rs]: Does this mean we can have void as a dev dependency only?

Comment (canndrew) [inline src/lib.rs]: It does! (Although it's a minuscule crate anyway)

File: src/utp_wrapper.rs [hunk 2]
```diff
@@ -56,6 +67,9 @@ impl UtpWrapper {
                                         Err(TryRecvError::Empty) => break,
                                     }
                                 }
+                                if send_keepalive {
+                                    socket.send_keepalive();
```
Comment (vinipsmaker) [inline src/utp_wrapper.rs]: This should be out of the write loop. If we only read for a long while, we will be silent and not send any keep alive.

**Comment by vinipsmaker** [comment]
It needs to be rebased. It contains commits that were merged in master already.

File: src/utp_wrapper.rs (hunk 2 above)
Comment (canndrew) [inline src/utp_wrapper.rs]: It is outside of the write loop. The only thing inside the write loop is `send_keepalive = false` which tells us not to bother with the keepalive if we've been writing.

Comment (vinipsmaker) [inline src/utp_wrapper.rs]: I got the identation wrong somehow. You're right.

File: Cargo.toml [hunk 3]
```diff
@@ -31,7 +31,6 @@ sodiumoxide = "~0.0.9"
 temp_utp = "~0.7.0"
```
Comment (vinipsmaker) [inline Cargo.toml]: update to "~0.8.0"

Comment (Fraser999) [inline Cargo.toml]: We have a few deps listed here without `~` - could that be fixed too please?

Comment (vinipsmaker) [inline Cargo.toml]: > We have a few deps listed here without `~` - could that be fixed too please?

This sounds like work for another PR. I filled an issue and will work on that soon: https://github.com/maidsafe/crust/issues/550

Comment (canndrew) [inline Cargo.toml]: Eh, I already fixed it.

**Comment by vinipsmaker** [comment]
Reviewed 2 of 4 files at r1, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/crust/543)*

**Pull request merged**